### ISSUE TITLE
Grad Clipping Not Working in Multi-Node Training with DeepSpeed Zero3 

### ISSUE THREAD

**Issue by hammoudhasan** [axolotl-ai-cloud/axolotl#1444]
### Please check that this issue hasn't been reported before.

- [X] I searched previous [Bug Reports](https://github.com/OpenAccess-AI-Collective/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

For the same learning rate if one reduces the value of `max_grad_norm` the model update should be tamed enough that the loss wouldn't spike and diverge.

### Current behaviour

Currently when I use an extremely low value of `max_grad_norm` such as `0.000001` the behavior is very similar to `1.0` hinting at the possibility that gradient clipping isn't working properly in my setting.

### Steps to reproduce

All you have to do is take any model you are training in a multi-node setting using `zero3_bf16.json` config and change the value of `max_grad_norm` and observe the loss behavior.

Note that I have taken a look at the issues and code of DeepSpeed and HuggingFace Trainer (from accelerator code) and I still can't find a possible solution.

### Which Operating Systems are you using?

- [X] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.10

### axolotl branch-commit

main/eef0072e9d652b9ebab75ae84326af90dd32e837

### Acknowledgements

- [X] My issue title is concise, descriptive, and in title casing.
- [X] I have searched the existing issues to make sure this bug has not been reported yet.
- [X] I am using the latest version of axolotl.
- [X] I have provided enough information for the maintainers to reproduce and diagnose the issue.

**Comment by winglian** [comment]
in accelerate it does this in `clip_grad_norm(...)`

```python
        elif self.distributed_type == DistributedType.DEEPSPEED:
            # `accelerator.backward(loss)` is doing that automatically. Therefore, its implementation is not needed
            # We cannot return the gradient norm because DeepSpeed does it.
            return None
```

if you look in your deepspeed json config, you'll likely see `gradient_clipping: auto`. Try adjusting it there.
